### PR TITLE
   Refine api

### PR DESCRIPTION
**Pull request by seiyanuta**

### CODE REVIEW

File: app/controllers/baseos_controller.rb [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+      logger.info "no deployments"
+      return nil
+    end
+
+    deployment
+  end
+
+
+  def auth_device
+    @device = Device.find_by_device_secret(params[:device_secret])
```
Comment (sh19910711) [inline app/controllers/baseos_controller.rb]: `image_params`?

File: app/controllers/apps_controller.rb [hunk 2]
```diff
@@ -2,63 +2,79 @@ class AppsController < ApplicationController
...
     build = Build.new
     build.status = 'queued'
-    build.app = App.find_by_name!(build_params[:name])
+    build.app = App.find_by_name!(build_params[:app_name])
     build.source_file = build_params[:source_file].read
     build.tag = build_params[:tag]
-    build.save!
+
+    unless build.save
+      resp :unprocessable_entity, { error: 'validation failed', reasons: app.errors.full_messages }
```
Comment (sh19910711) [inline app/controllers/apps_controller.rb]: `return` statement

File: app/controllers/baseos_controller.rb (hunk 1 above)
Comment (seiyanuta) [inline app/controllers/baseos_controller.rb]: This method is a `before_action`. `image_params` is defined for `image` method.

File: app/controllers/apps_controller.rb (hunk 2 above)
Comment (seiyanuta) [inline app/controllers/apps_controller.rb]: fixed

File: app/controllers/baseos_controller.rb (hunk 1 above)
Comment (sh19910711) [inline app/controllers/baseos_controller.rb]: :+1:

File: app/controllers/devices_controller.rb [hunk 3]
```diff
@@ -16,131 +15,27 @@ def create
...
 
-    render json: { device_secret: device.device_secret }
+    if device.save
+      resp :ok, { device_secret: device.device_secret }
+    else
+      resp :unprocessable_entity, { error: 'validation failed', reasons: device.errors.full_messages }
+    end
   end
 
   def update
```
Comment (sh19910711) [inline app/controllers/devices_controller.rb]: The method should be moved into the baseos controller?

File: app/controllers/devices_controller.rb [hunk 4]
```diff
@@ -16,131 +15,27 @@ def create
...
-    device = current_team.devices.find_by_device_secret!(device_params[:device_secret])
+    device = current_team.devices.find_by_device_secret!(params[:device_secret])
     device.update_attributes(devce_params)
```
Comment (sh19910711) [inline app/controllers/devices_controller.rb]: typo in the argument

**Comment by sh19910711** [comment]
Seems good :+1: Thank you.

**Pull request merged**